src/components/drive-container.tsx

```diff
@@ -0,0 +1,136 @@
+import {
+    ActionType,
+    DriveTreeItem,
+    DriveView,
+    ItemType,
+    OnItemOptionsClickHandler,
+    TreeItem,
+    traverseTree,
+} from '@powerhousedao/design-system';
+import { Drive, Node } from 'document-model-libs/document-drive';
+import { Immutable } from 'document-model/document';
+import { useEffect, useState } from 'react';
+import { useDocumentDrive } from 'src/hooks/useDocumentDrive';
+
+function mapDocumentDriveNodeToTreeItem(
+    node: Immutable<Node>,
+    allNodes: Immutable<Array<Node>>
+): TreeItem {
+    const isFolder = node.kind === 'folder';
+    const pathRegex = new RegExp(`^${node.path}/[^/]+`);
+
+    return {
+        id: node.path,
+        label: node.name,
+        type: isFolder ? ItemType.Folder : ItemType.File,
+        children: isFolder
+            ? allNodes
+                  .filter(childrenNode => pathRegex.test(childrenNode.path))
+                  .map(childrenNode =>
+                      mapDocumentDriveNodeToTreeItem(childrenNode, allNodes)
+                  )
+            : undefined,
+    };
+}
+
+function mapDocumentDriveToTreeItem(drive: Immutable<Drive>): DriveTreeItem {
+    return {
+        id: drive.id,
+        label: drive.name,
+        type: ItemType.LocalDrive,
+        expanded: true,
+        children: drive.nodes
+            .filter(node => !node.path.includes('/'))
+            .map(node => mapDocumentDriveNodeToTreeItem(node, drive.nodes)),
+    };
+}
+
+export default function () {
+    const [drives, setDrives] = useState<DriveTreeItem[]>([]);
+    const { documentDrive, openFile, addFolder, deleteNode, renameNode } =
+        useDocumentDrive();
+
+    useEffect(() => {
+        const newItems = documentDrive?.state.drives.map(
+            mapDocumentDriveToTreeItem
+        );
+        setDrives(newItems ?? []);
+    }, [documentDrive]);
+
+    function handleNodeClick(item: TreeItem, drive: DriveTreeItem) {
+        if (item.type === ItemType.File) {
```

Comment:
here we should handle open or close the folder:

```
    function handleNodeClick(item: TreeItem, drive: DriveTreeItem) {
        if (item.type === ItemType.File) {
            openFile(drive.id, item.id);
        } else {
            setDrives(drives =>
                traverseDriveById(drives, drive.id, treeItem => {
                    if (treeItem.id === item.id) {
                        return {
                            ...treeItem,
                            expanded: !treeItem.expanded,
                        };
                    }

                    return treeItem;
                })
            );
        }
    }
            
```

Reply:
I've added this implementation in this [PR](https://github.com/powerhouse-inc/document-model-electron/pull/10)